src/stravalib/unithelper.py

```diff
@@ -4,84 +4,43 @@
 Helpers for converting Strava's units to something more practical.
 """
 
-from numbers import Number
-from typing import Any, Protocol, Union, cast, runtime_checkable
+from typing import Any, Union
 
 import pint
+from pint.facets.plain import PlainQuantity
 
-from stravalib.exc import warn_units_deprecated
-from stravalib.unit_registry import Q_
+from stravalib.unit_registry import ureg
 
 
-@runtime_checkable
-class UnitsQuantity(Protocol):
+class _Quantity(float):
     """
-    A type that represents the (deprecated) `units` Quantity. The `unit`
-    attribute in the units library consists of other classes, so this
-    representation may not be 100% backward compatible!
+    Subtype of float that can represent quantities by adding a unit
     """
 
-    num: float
     unit: str
 
+    def quantity(self) -> pint.Quantity:
+        return ureg.Quantity(self, self.unit)
 
-class Quantity(Q_):  # type: ignore[valid-type, misc]
+
+class UnitConverter:
     """
-    Extension of `pint.Quantity` for temporary backward compatibility with
-    the legacy `units` package.
+    Callable that converts quantities or unitless numbers to quantities of its unit
     """
 
-    @property
-    def num(self) -> pint._typing.Magnitude:
-        warn_units_deprecated()
-        return self.magnitude
-
-    @property
-    def unit(self) -> str:
-        warn_units_deprecated()
-        return str(self.units)
-
-    def __int__(self) -> int:
-        return int(self.magnitude)
-
-    def __float__(self) -> float:
-        return float(self.magnitude)
-
-
-class UnitConverter:
     def __init__(self, unit: str) -> None:
         self.unit = unit
 
     def __call__(
-        self, q: Union[Number, pint.Quantity, UnitsQuantity]
-    ) -> Quantity:
-        if isinstance(q, Number):
-            # provided quantity is unitless, so mimic legacy `units` behavior:
-            converted_q = Quantity(q, self.unit)
+        self, q: Union[_Quantity, pint.Quantity, float]
+    ) -> PlainQuantity[Any]:
+        if isinstance(q, pint.Quantity):
+            return q.to(self.unit)
+        elif isinstance(q, _Quantity):
+            return q.quantity().to(self.unit)
         else:
-            try:
-                converted_q = Quantity(
-                    cast(pint.Quantity, q).to(self.unit).magnitude, self.unit
-                )
-            except AttributeError:
-                # unexpected type of quantity, maybe it's a legacy `units` Quantity
-                warn_units_deprecated()
-                computed_q = Quantity(q.num, q.unit)
-                converted_q = Quantity(
-                    computed_q.to(self.unit).magnitude, self.unit
-                )
-
-        return converted_q
-
-
-def is_quantity_type(obj: Any) -> bool:
-    if isinstance(obj, (pint.Quantity, Quantity)):
-        return True
-    elif isinstance(obj, UnitsQuantity):  # check using Duck Typing
-        warn_units_deprecated()
-        return True
-    else:
-        return False
```

Comment:
This is not needed anymore because we drop backward compatibility with the `units` package.